### PR TITLE
Update availability/platform badge

### PR DESCRIPTION
**Pull request by lshep**
- instead of convoluted combo of building and .bbsoption
    display reflection of .bbsoption and deprecation status

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -448,29 +448,64 @@ end
...
   site_config = YAML.load_file("./config.yaml")
   for reldev in ['release', 'devel']
+    puts "Working on #{reldev}"
     numeric_version = (reldev == 'release') ? site_config['release_version'] : site_config['devel_version']
-    unsupported_platforms = {}
     meat_index_file = File.join("tmp", "build_dbs","#{reldev}-bioc.meat-index.txt")
-    json_file = File.join("assets", "packages", "json", numeric_version, "bioc",
-      "packages.json")
+    json_file = File.join("assets", "packages", "json", numeric_version, "bioc", "packages.json")
+    json_obj = JSON.parse(File.read(json_file))
```
Comment (mtmorgan) [inline Rakefile]: for review clarity, I would have left L456, 7 where they were...

File: Rakefile [hunk 2]
```diff
@@ -448,29 +448,64 @@ end
...
+          if status == "Deprecated"
+            img = "none"
+          else
+            if unsupported == "None"
+              img = "all"
+            else
+              img = "some"
+            end
+          end
+          puts "Creating badge for #{pkg} :  #{img}"
```
Comment (mtmorgan) [inline Rakefile]: Seems like this code chunk could be a function and reused a few lines down?

File: Rakefile [hunk 3]
```diff
@@ -448,29 +448,64 @@ end
...
+    indexList=[]
     if File.exists? meat_index_file
       mitxt = File.readlines(meat_index_file).join
       meat_index = Dcf.parse(mitxt)
-      json_obj = JSON.parse(File.read(json_file))
       if (not meat_index.nil?)
-	for item in meat_index
-	  view = json_obj[item['Package']]
-	  next if view.nil?
-	  get_available(item, numeric_version, view)
```
Comment (mtmorgan) [inline Rakefile:467]: generally, I like this design better, where a function encapsulates an action. This makes it easier to think about isolated operations in the code, including testing the specific functionality without having to process all badges...

**Comment by lshep** [comment]
@mtmorgan   the code has been updated to separate out into helper functions.  Please re-review at earliest convenience

File: Rakefile [hunk 4]
```diff
@@ -454,49 +454,27 @@ task :get_availability_shields  do
...
+	end
+	unknown = json_obj.keys.sort - indexList.sort
+	if unknown.length != 0
+	  for item in unknown
+	    availabilityBadge(item, "unknown-build", numeric_version)
+	  end
+	end
       else
-        puts "ERROR meat_index nil"
+	puts "ERROR meat_index nil"
```
Comment (mtmorgan) [inline Rakefile:477]: sorry to be a pain but would rather not see the whitespace changes here and below...

**Review by mtmorgan — APPROVED** [review]

File: Rakefile (hunk 4 above)
Comment (lshep) [reply Rakefile:477]: how do I change back to a whitespace-cleanup call without losing the changes?

**Pull request merged**